include/deal.II/base/tensor.h

```diff
@@ -687,11 +728,13 @@ class Tensor
 
   /**
    * This constructor is for internal use. It provides a way
-   *  to create constexpr constructors for Tensor<rank, dim, Number>
+   * to create constexpr constructors for Tensor<rank, dim, Number>
+   *
+   * @note This function can also be used in CUDA device code.
    */
   template <typename ArrayLike, std::size_t... Indices>
-  constexpr Tensor(const ArrayLike &initializer,
-                   std_cxx14::index_sequence<Indices...>);
+  constexpr DEAL_II_CUDA_HOST_DEV
+  Tensor(const ArrayLike &initializer, std_cxx14::index_sequence<Indices...>);
```

Comment:
Does that work with CUDA 8.0? There is no support for c++14 in CUDA 8.0 and we can't fall back on boost...

Reply:
We can test that.

Reply:
The test also passes using `CUDA 8.0`.

Reply:
In the end, `std_cxx14::index_sequence` doesn't even use `boost`.